gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -208,18 +214,20 @@ public GobblinServiceJobScheduler(@Named(InjectionNames.SERVICE_NAME) String ser
       metricContext.register(this.totalAddSpecTimeNanos);
       metricContext.register(this.numJobsScheduledDuringStartup);
     }
+    this.dagProcessingEngineEnabled = ConfigUtils.getBoolean(config, ServiceConfigKeys.DAG_PROCESSING_ENGINE_ENABLED, false);
```

Comment:
I keep seeing classes looking into config to find out this value.  is there really no way to have guice calculate/read it once and then pass the value to each of these other classes it's already involved in initializing?